### PR TITLE
Softsign optimization

### PR DESCRIPTION
**Pull request by nemerchiedde**
Storing only the positive part of the Softsign is possible due to its symmetry in the Y axis. Same theory when presented in pull request: https://github.com/fastmachinelearning/hls4ml/pull/527

### CODE REVIEW

**Comment by bo3z** [comment]
Looks good to me. Would be nice to add a test, just to show correct behaviour. We could do it in a similar way to other activations, in `test/test_keras_api.py`

**Comment by nemerchiedde** [comment]
> Looks good to me. Would be nice to add a test, just to show correct behaviour. We could do it in a similar way to other activations, in `test/test_keras_api.py`

Added one test file.

File: test/pytest/test_softsign.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    hls_model.compile()
+   
+    y_keras = model.predict(X)
+    y_hls4ml = hls_model.predict(X).reshape(y_keras.shape)
+
+    acc_hls4ml = accuracy_score(np.argmax(y_keras, axis=-1).ravel(), np.argmax(y_hls4ml, axis=-1).ravel())
+
+    print('Accuracy hls4ml relative to keras: {}'.format(acc_hls4ml))
+
+    assert acc_hls4ml >= 0.7
```
Comment (vloncar) [inline test/pytest/test_softsign.py:56]: Why only 0.7?

Comment (nemerchiedde) [reply test/pytest/test_softsign.py]: Changed to 0.98. New tests have also been added to the pytest file.

**Review by vloncar — APPROVED** [review]
